_deploy/r/gnoswap/common/access.gno

```diff
@@ -23,56 +27,65 @@ func SatisfyCond(cond bool) error {
 
 func AdminOnly(caller std.Address) error {
 	if caller != consts.ADMIN {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+		return ufmt.Errorf(ErrNoPermission, caller.String())
 	}
 	return nil
 }
 
 func GovernanceOnly(caller std.Address) error {
 	if caller != consts.GOV_GOVERNANCE_ADDR {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+		return ufmt.Errorf(ErrNoPermission, caller.String())
 	}
 	return nil
 }
 
 func GovStakerOnly(caller std.Address) error {
 	if caller != consts.GOV_STAKER_ADDR {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+		return ufmt.Errorf(ErrNoPermission, caller.String())
 	}
 	return nil
 }
 
 func RouterOnly(caller std.Address) error {
 	if caller != consts.ROUTER_ADDR {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+		return ufmt.Errorf(ErrNoPermission, caller.String())
 	}
 	return nil
 }
 
 func PositionOnly(caller std.Address) error {
 	if caller != consts.POSITION_ADDR {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+		return ufmt.Errorf(ErrNoPermission, caller.String())
 	}
 	return nil
 }
 
 func StakerOnly(caller std.Address) error {
 	if caller != consts.STAKER_ADDR {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+		return ufmt.Errorf(ErrNoPermission, caller.String())
 	}
 	return nil
 }
 
-func TokenRegisterOnly(caller std.Address) error {
-	if caller != consts.TOKEN_REGISTER {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+func LaunchpadOnly(caller std.Address) error {
+	if caller != consts.LAUNCHPAD_ADDR {
+		return ufmt.Errorf(ErrNoPermission, caller.String())
 	}
 	return nil
 }
 
-func LaunchpadOnly(caller std.Address) error {
-	if caller != consts.LAUNCHPAD_ADDR {
-		return ufmt.Errorf("caller(%s) has no permission", caller.String())
+func EmissionOnly(caller std.Address) error {
+	if caller != consts.EMISSION_ADDR {
+		return ufmt.Errorf(ErrNoPermission, caller.String())
+	}
+	return nil
+}
+
+// DEPRECATED
+// TODO: remove after r/grc20reg is applied for all contracts
+func TokenRegisterOnly(caller std.Address) error {
```

Comment:
Remove?

Reply:
If removing the old registry doesn't have much side effect across contracts, replace it at once, not by a contract.

Reply:
currently, old register has been removed only pool contract.

I'll make new pr that remove entire old registe logicr from all contract.